dandi/dandiarchive.py

```diff
@@ -890,9 +890,25 @@ def follow_redirect(url: str) -> str:
             or one of the statuses in `~dandi.consts.RETRY_STATUSES` is returned
         """
         i = 0
+        retries = 4
         while True:
```

Comment:
Since we are going for minimal changes, this is fine IMO, but  `for i in range(retries + 1):` would be more idiomatic and would simplify a bit.

Reply:
it seems this should indeed work now -- I think there was a stage in code at some point that we could fall through the loop body without returning or raising... looking at it now - indeed I think we could just switch  to for loop now, thanks!